src/west/app/main.py

```diff
@@ -860,21 +841,61 @@ def add_argument(self, *args, **kwargs):
         super().add_argument(*args, **kwargs)
 
     def error(self, message):
-        if (self.west_app and
-                self.west_app.mle and
-                isinstance(self.west_app.mle,
-                           ManifestVersionError) and
-                self.west_app.cmd):
-            self.west_app.cmd.die(mve_msg(self.west_app.mle))
+        if self.west_app and self.west_app.mle:
+            # If we have a known WestApp instance and the manifest
+            # failed to load, then try to specialize the error message
+            # to handle west-specific situations.
+
+            app = self.west_app
+            mle = self.west_app.mle
+            if app.cmd:
+                cmd = app.cmd
```

Comment:
Maybe `failed_cmd = app.cmd`? I've been experiencing a fair amount of "shorthand" fatigue this review...

Reply:
> Maybe failed_cmd = app.cmd?

That's not accurate, though: we are just looking for a WestCommand we can call .die() on. That might be a command which has failed, but it also might be the case that we have no knowledge of what command to run, and are just grabbing some other WestCommand instance, somewhat at random, to use instead. This method is definitely a white-box sort of thing, relying on implementation details elsewhere.